app/forms/punches_create_form.rb

```diff
@@ -0,0 +1,46 @@
+class PunchesCreateForm
+  include ActiveModel::Model
+
+  attr_accessor :punches
+
+  validates :punches, presence: true
+  validate :punches_validations, :punch_on_same_day_validation
+
+  def initialize(params)
+    params ||= {}
+    params[:punches] ||= []
+    params[:user] ||= User.new
+
+    @punches = params[:punches].map do |punch_params|
+      Punch.new(**punch_params, user: params[:user], company: params[:user].company)
+    end
+  end
+
+  def save
+    return false unless valid?
+
+    @punches = Punch.transaction do
+      @punches.each(&:save!)
+    end
```

Comment:
[Question] Should this really be assigning to `@punches`?

Reply:
I thought that I needed that to reload, for example the id, but I don't need, thanks for the observation.